test/cube-dimension-simplified.js

```diff
@@ -0,0 +1,201 @@
+                new TimeDimension('time', 'month', '2010-01', '2010-02'),
+            ]);
+
+            cube.createStoredMeasure('measure1', {}, 'float32', 0);
+
+            assert.deepEqual(cube.getNestedObject('measure1'), {
+                paris: { '2010-01': 0, '2010-02': 0 },
+                madrid: { '2010-01': 0, '2010-02': 0 },
+                berlin: { '2010-01': 0, '2010-02': 0 },
+            });
+
+            assert.deepEqual(cube.removeDimension('location').getNestedObject('measure1'), {
+                '2010-01': 0,
+                '2010-02': 0,
+            });
+        });
+
+        it('should be able to remove a dimension with some values in it', function () {
+            const newDimension = new GenericDimension('location', 'root', [
+                'paris',
+                'madrid',
+                'berlin',
+            ]);
+            const cube = new Cube([
+                newDimension,
+                new TimeDimension('time', 'month', '2010-01', '2010-02'),
+            ]);
+
+            cube.createStoredMeasure('measure1', {}, 'float32', 0);
+
+            cube.hydrateFromSparseNestedObject('measure1', {
+                paris: { '2010-01': 10, '2010-02': 0 },
+                madrid: { '2010-01': 0, '2010-02': 5 },
+                berlin: { '2010-01': 0, '2010-02': 10 },
+            });
+
+            // assert.deepEqual(cube.getNestedObject('measure1'), {
+            //     paris: { '2010-01': 0, '2010-02': 0 },
+            //     madrid: { '2010-01': 0, '2010-02': 0 },
+            //     berlin: { '2010-01': 0, '2010-02': 0 },
+            // });
+
+            assert.deepEqual(cube.removeDimension('location').getNestedObject('measure1'), {
+                '2010-01': 10,
+                '2010-02': 15,
+            });
+        });
+
+        // it('should be able to add a time dimension', function () {
+        //     const cube = new Cube([new TimeDimension('time1', 'month', '2010-01', '2010-02')]);
+        //     cube.createStoredMeasure('measure1', { time: 'sum' }, 'float32', 100);
+        //     cube.createStoredMeasure('measure2', { time: 'average' }, 'float32', 100);
+
+        //     const newDimension = new TimeDimension(
+        //         'time2',
+        //         'week_mon',
+        //         '2010-W01-mon',
+        //         '2010-W08-mon'
+        //     );
+        //     const newCube = cube.addDimension(newDimension, {
+        //         measure1: 'sum',
+        //         measure2: 'average',
+        //     });
+
+        //     assert.deepEqual(
+        //         newCube.removeDimension('time2').getNestedObject('measure1'),
+        //         cube.getNestedObject('measure1')
+        //     );
+
+        //     assert.deepEqual(
+        //         newCube.removeDimension('time2').getNestedObject('measure2'),
+        //         cube.getNestedObject('measure2')
+        //     );
+        // });
+    });
+
+    // describe('removeDimension', function () {
```

Comment:
We need to re enable those, right?

Reply:
:+1: I've merged the two dimension test files together now.